portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,12 +48,14 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query(ENTITY_NAME).addSort(PROPERTY_FACT, SortDirection.DESCENDING);
 
+    int commentsLimit = Integer.parseInt(getParameter(request, "limit", "100"));
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     // Add previously inputted fun facts to userFacts
     ImmutableList<String> userFacts =
-        Streams.stream(results.asIterable())
+        Streams.stream(results.asIterable(FetchOptions.Builder.withLimit(commentsLimit)))
```

Comment:
nit: maybe it's simpler and easier to read to use the streams .limit(commentsLimit) (you can add this after this line) operator instead of applying it on the iterable

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#limit-long-

Reply:
Oh okay, that definitely makes it easier. Thanks!